### PR TITLE
[MU4] New score wizard. Instruments page refactoring

### PR DESCRIPTION
**Pull request by RomanPudashkin**
Resolves:
https://github.com/musescore/MuseScore/issues/7639
https://github.com/musescore/MuseScore/issues/8698
https://github.com/musescore/MuseScore/issues/8734

### CODE REVIEW

File: src/instrumentsscene/qml/MuseScore/InstrumentsScene/internal/InstrumentsOnScoreView.qml [hunk 1]
```diff
@@ -30,50 +30,47 @@ import MuseScore.InstrumentsScene 1.0
...
-    signal orderChanged(string id)
-    signal soloistChanged(string id)
+    function addInstruments(instruments) {
+        instrumentsOnScore.addInstruments(instruments)
+    }
 
-    function selectedScoreOrder() {
-        var orders = instrumentsModel.scoreOrders
-        return orders[scoreOrderComboBox.currentIndex].config
+    function moveInstrumentsUp() {
```
Comment (Eism) [inline src/instrumentsscene/qml/MuseScore/InstrumentsScene/internal/InstrumentsOnScoreView.qml:51]: moveSelectedInstrumentsUp

**Comment by RobFog** [comment]
Will you consider #8871 also? I believe @Tantacrul acknowledged it is a valid issue.

**Comment by RomanPudashkin** [comment]
> Will you consider #8871 also? I believe @Tantacrul acknowledged it is a valid issue.

Sure, I'll investigate the performance issue, but later. It isn't related to this PR, because my PR is mostly about the new score wizard

File: src/instrumentsscene/qml/MuseScore/InstrumentsScene/internal/SelectedInstrumentsView.qml [hunk 2]
```diff
@@ -30,45 +30,38 @@ import MuseScore.InstrumentsScene 1.0
 Item {
     id: root
 
-    property alias instruments: instrumentsView.model
-    property var instrumentsModel: null
+    property bool hasInstruments: instrumentsOnScore.count > 0
+    property alias isMovingUpAvailable: instrumentsOnScore.isMovingUpAvailable
+    property alias isMovingDownAvailable: instrumentsOnScore.isMovingDownAvailable
```
Comment (cbjeukendrup) [inline src/instrumentsscene/qml/MuseScore/InstrumentsScene/internal/SelectedInstrumentsView.qml:35]: How about making these properties `readonly`?

**Comment by Jojo-Schmitz** [comment]
rebase needed

File: src/instrumentsscene/qml/MuseScore/InstrumentsScene/internal/SelectedInstrumentsView.qml (hunk 2 above)
Comment (RomanPudashkin) [reply src/instrumentsscene/qml/MuseScore/InstrumentsScene/internal/SelectedInstrumentsView.qml]: fixed

File: src/instrumentsscene/qml/MuseScore/InstrumentsScene/internal/InstrumentsOnScoreView.qml (hunk 1 above)
Comment (RomanPudashkin) [reply src/instrumentsscene/qml/MuseScore/InstrumentsScene/internal/InstrumentsOnScoreView.qml]: fixed

**Comment by vshalkevich** [comment]
Works good

File: src/instrumentsscene/view/instrumentlistmodel.cpp [hunk 3]
```diff
@@ -272,373 +256,233 @@ void InstrumentListModel::sortInstruments(QVariantList& instruments) const
...
+        } else {
+            doSetCurrentGroup(NONE_GROUP_ID);
+        }
     }
 
-    m_searchText = text;
-    emit dataChanged();
-
-    updateGenreStateBySearch();
+    emit dataChanged(index(0), index(rowCount() - 1), { RoleIsSelected });
```
Comment (Eism) [inline src/instrumentsscene/view/instrumentlistmodel.cpp:296]: Can you notify about the selected state changed only for the current instrument(modelIndex)?

File: src/notation/notationtypes.h [hunk 4]
```diff
@@ -309,11 +311,9 @@ struct PitchRange
     }
 };
 
-static const QString COMMON_GENRE_ID("common");
-
 struct InstrumentKey
 {
-    ID instrumentId;
+    QString instrumentId;
```
Comment (Eism) [inline src/notation/notationtypes.h:316]: can you explain why you replaced to QString?

**Review by Eism — APPROVED** [review]

**Pull request merged**